Feature/#152 document technology and variants (#157)
ackama/rails-template: README.md

generator system, allowing all of its ERb templates and files to be referenced
when the application template script is evaluated.

Rails generators are very lightly documented; what you’ll find is that most of the heavy lifting is done by  [Thor](http://whatisthor.com/). Thor is a tool that allows you to easily perform command line utilities.  The most common methods used by this template are Thor’s `copy_file`, `template`, and `gsub_file`. You can dig into the well-organized and well-documented [Thor source code](https://github.com/erikhuda/thor) to learn more.
If any file finishes with `.tt`, Thor considers it to be a template and places it in the destination without the extension `.tt`. 

## Tooling choices and configuration
  
#### Testing: [webdrivers](https://github.com/titusfortner/webdrivers)
auto-installs headless Chrome

#### ENV managment: [dotenv](https://github.com/bkeepers/dotenv)  
in place of the Rails `secrets.yml`
  
#### Application web server: [puma](https://github.com/puma/puma)
  

> #### Security Auditing: [brakeman](https://github.com/presidentbeef/brakeman) and [bundler-audit](https://github.com/rubysec/bundler-audit)

Reviewer: We also use [`yarn audit`](https://classic.yarnpkg.com/en/docs/cli/audit/)/[`npm audit`](https://docs.npmjs.com/cli/audit) for auditing js dependencies.

Author: added in a mention of yarn audit
https://github.com/ackama/rails-template/pull/157/commits/d2bc43d5833fb98ea21a7c3ad9631827a3a1974c

“EditorConfig helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. The EditorConfig project consists of a file format for defining coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. EditorConfig files are easily readable and they work nicely with version control systems” 

See the `editorconfig` file for Ackama’s styles

#### Git hook manager: [Overcommit](https://github.com/sds/overcommit) 
Git has a way to fire off custom scripts when certain important actions occur, by using [Git hooks ](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)

Overcommit is a tool to manage and configure Git hooks

Edit `overcommit.yml` to configure the Overcommit hooks you wish to use

#### Code coverage measurement: [Simplecov](https://github.com/simplecov-ruby/simplecov)
Code coverage is a measurement of how many lines/blocks/arcs of your code are executed while the automated tests are running.

SimpleCov is a code coverage analysis tool for Ruby. It provides an API to filter, group, merge, format, and display the results

> #### JS package manager: [Yarn](https://yarnpkg.com/)

Reviewer: ```suggestion
#### JS package manager: [Yarn](https://classic.yarnpkg.com/lang/en/)
```

We're not using Yarn 2 - it's probably worth noting that explicitly too, but that might make things a bit more confusing.

Generally a standard yarn install (i.e `npm i -g yarn`) should always give you Yarn 1.

> “Yarn is a package manager for your code”. Think of it as Bundler for JavaScript. Yarn uses a file similar to Bundler’s Gemfile called `package.json`, found in the Rails root directory.

Reviewer 2: @G-Rath I know we've discussed in the past that we're getting close to a time where switching to NPM as our default would make sense. We'll need to remember to update this part of the readme if we do

Reviewer: Yeah, I think it would be worth mentioning `npm` here as most ecosystems only have the one package manager, whereas JS has three.

Also, `npm` 7 (which will be landing soon) has support for `yarn.lock` which'll make things easier.

Reviewer: Also also,

```suggestion
“Yarn is a package manager for your code”. Think of it as Bundler for JavaScript. Javascript dependencies are tracked in a file similar to Bundler’s Gemfile called `package.json`, found in the Rails root directory. JS package managers such as Yarn use this to determine what to install, and record this in a lock file.
```

I think a phrasing like this is a bit nicer because it makes it a bit clearer that `package.json` isn't a Yarn-specific thing, while still keeping a balance between "factually accurate" and "what you do/don't need to know for this to work".

#### Code linting and formatting
Linting is the automated checking of your source code for programmatic and stylistic errors. Check the docs of each linter for how to automatically format code. 

> - JS/HTML/CSS:  [Prettier](https://prettier.io/), Set up with  an [Ackama prettier config](https://github.com/ackama/prettier-config-ackama) and a variety of other prettier plugins, see the full list in `variants/frontend-base/template.rb`

Reviewer: ```suggestion
- JS/HTML/CSS: [Prettier](https://prettier.io/), Set up with an [Ackama prettier config](https://github.com/ackama/prettier-config-ackama) and a variety of other prettier plugins, see the full list in `variants/frontend-base/template.rb`
```

Prettier also does markdown docs ;)

--------

🐘ish - the plugins we install are all for `eslint`, of which `prettier` is one :)
